### ISSUE TITLE
When restricting cards to only LTR and LTC, sometimes cards from other sets are shown

### ISSUE THREAD

**Issue by sargreal** [sargreal/forge#1]

**Comment by Fhavyre** [comment]
Tales of Middle-earth Commander (LTC) does have reprints in the set. The only thing that can be considered a bug in this context is that we want to select the new art instead of the originals

**Comment by sargreal** [comment]
You are right that the label bug is not perfect, but I used it as it is a problem with the current configuration or the possibilities of the implementation

So the current filter looks like this: 
https://github.com/sargreal/forge/blob/middleearth/forge-gui/res/adventure/MiddleEarth/config.json#L40-L48

We talked about the reprints, however I would expect that Cards are removed/selected based on their ID and the association with a set.

With the current behavior however I suspect that it goes the way: 

`Edition Legal -> Card Name Legal -> Card ID Legal`

but I would expect:

`Edition Legal -> Card ID Legal`

Additionally it is a problem that restricting all editions does not restrict old cards that do not have an edition. So we should probably add a "*" filter for `restrictedCards` if that does not exist yet

**Comment by sargreal** [comment]
Found the issue, there are 3 parts to this:

- There is a config option `useAllCardVersions`. This allows for multiple different card versions. Wihtout this enabled, reprints will most likely be other versions
- when the mentioned option is disabled, always the first matching card is selected, not really what we want
- when it is enabled, all different versions are possible, also not what we want

I am already implementing a fix for this, will push it later.

Also I have found that the restricted `(ALL)` does not do anything.
